src/storage/btree_load.c

```diff
@@ -4903,15 +4917,17 @@ online_index_builder (THREAD_ENTRY * thread_p, BTID_INT * btid_int, HFID * hfids
 	}
 
       /* Dispatch the insert operation */
-      index_builder_loader_task *load_task = new index_builder_loader_task (btid_int->sys_btid, &class_oids[cur_class],
-									    &cur_oid, unique_pk, load_context);
-
-      load_task->set_key (p_dbvalue);
-
-      thread_get_manager ()->push_task (ib_workpool, load_task);
-
-      /* Increment tasks started. */
-      tasks_started++;
+      if (load_task == NULL)
```

Comment:
The code wants to collect items into index_builder_loader_task, and start a thread when the memory threshold is reached. For readability reason:
- add proper comment
- replace code like thread_get_manager ()->push_task (ib_workpool, load_task.release ()), with:
   - task = load_task.release (); assert (load_tsk.get() == NULL);) ;thread_get_manager ()->push_task (ib_workpool, task);
 Similar for load_task.reset (new index_builder_loader_task (btid_int->sys_btid, &class_oids[cur_class], unique_pk,
                                                          load_context));